bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/handler/DeviceHandler.java

```diff
@@ -0,0 +1,345 @@
+    private final Logger logger;
+    @NonNullByDefault({})
+    private String dsn;
+    @NonNullByDefault({})
+    private CloudApi cloudApi;
+    private final Map<String, String> channelUidMap = new HashMap<>();
+    private final Map<String, String> channelX100UidMap = new HashMap<>();
+
+    public DeviceHandler(Thing thing) {
+        super(thing);
+        logger = LoggerFactory.getLogger(DeviceHandler.class.getName() + "[" + thing.getUID().getId() + "]");
+    }
+
+    @Override
+    public void initialize() {
+        try {
+            internalInitialize();
+        } catch (Exception e) {
+            updateStatus(OFFLINE, CONFIGURATION_ERROR,
+                    "Error occurred while initializing Salus device! " + e.getLocalizedMessage());
+        }
+    }
+
+    private void internalInitialize() {
+        var bridge = getBridge();
+        if (bridge == null) {
+            updateStatus(OFFLINE, BRIDGE_UNINITIALIZED,
+                    "There is no bridge for this thing. Remove it and add it again.");
+            return;
+        }
+        var bridgeHandler = bridge.getHandler();
+        if (!(bridgeHandler instanceof CloudBridgeHandler cloudHandler)) {
+            updateStatus(OFFLINE, BRIDGE_UNINITIALIZED, "There is wrong type of bridge for cloud device!");
+            return;
+        }
+        this.cloudApi = cloudHandler;
+
+        dsn = (String) getConfig().get(DSN);
+
+        if ("".equals(dsn)) {
+            updateStatus(OFFLINE, CONFIGURATION_ERROR,
+                    "There is no " + DSN + " for this thing. Remove it and add it again.");
+            return;
+        }
+
+        try {
+            var device = this.cloudApi.findDevice(dsn);
+            if (device.isEmpty()) {
+                var msg = "Device with DSN " + dsn + " not found!";
+                updateStatus(OFFLINE, COMMUNICATION_ERROR, msg);
+                return;
+            }
+            if (!device.get().isConnected()) {
+                var msg = "Device with DSN " + dsn + " is not connected!";
+                updateStatus(OFFLINE, COMMUNICATION_ERROR, msg);
+                return;
+            }
+            var channels = findDeviceProperties().stream().map(this::buildChannel).toList();
+            if (channels.isEmpty()) {
+                updateStatus(OFFLINE, CONFIGURATION_ERROR, "There are no channels for " + dsn + ".");
+                return;
+            }
+            updateChannels(channels);
+        } catch (Exception e) {
+            var msg = "Error when connecting to Salus Cloud!";
+            updateStatus(OFFLINE, COMMUNICATION_ERROR, msg);
+            return;
+        }
+
+        // done
+        updateStatus(ONLINE);
+    }
+
+    private Channel buildChannel(DeviceProperty<?> property) {
+        String channelId;
+        String acceptedItemType;
+        if (property instanceof DeviceProperty.BooleanDeviceProperty booleanProperty) {
```

Comment:
```suggestion
        if (property instanceof DeviceProperty.BooleanDeviceProperty) {
```

`booleanProperty` is not used withiin this block, can be removed.